tests/integration/src/dfx_orbit/manage_settings.rs

```diff
@@ -0,0 +1,159 @@
+    submit_request(
+        &env,
+        WALLET_ADMIN_USER,
+        canister_ids.station,
+        RequestOperationInput::CreateExternalCanister(CreateExternalCanisterOperationInput {
+            kind: CreateExternalCanisterOperationKindDTO::AddExisting(
+                CreateExternalCanisterOperationKindAddExistingDTO { canister_id },
+            ),
+            name: String::from("counter"),
+            description: None,
+            labels: None,
+            permissions: ExternalCanisterPermissionsCreateInput {
+                calls: vec![],
+                read: AllowDTO {
+                    auth_scope: station_api::AuthScopeDTO::Restricted,
+                    user_groups: vec![],
+                    users: vec![],
+                },
+                change: AllowDTO {
+                    auth_scope: station_api::AuthScopeDTO::Restricted,
+                    user_groups: vec![],
+                    users: vec![],
+                },
+            },
+            request_policies: ExternalCanisterRequestPoliciesCreateInput {
+                change: vec![],
+                calls: vec![],
+            },
+        }),
+    );
+
+    let config = DfxOrbitTestConfig {
+        canister_ids: vec![(String::from("counter"), canister_id)],
+        ..Default::default()
+    };
+
+    let add_controller_args = RequestCanisterUpdateSettingsArgs {
+        canister: String::from("counter"),
+        add_controller: vec![user_test_id(1)],
+        remove_controller: vec![],
+    };
+
+    let add_remove_controller_args = RequestCanisterUpdateSettingsArgs {
+        canister: String::from("counter"),
+        add_controller: vec![user_test_id(2)],
+        remove_controller: vec![user_test_id(1)],
+    };
+
+    let remove_controller_args = RequestCanisterUpdateSettingsArgs {
+        canister: String::from("counter"),
+        add_controller: vec![],
+        remove_controller: vec![canister_ids.station],
+    };
+
+    dfx_orbit_test(&mut env, config, async {
+        // Setup the station agent
+        let dfx_orbit = setup_dfx_orbit(canister_ids.station).await;
+
+        let controllers = dfx_orbit.get_controllers(canister_id).await.unwrap();
+        assert_eq!(controllers, vec![canister_ids.station]);
+
+        // Add another controller to the test canister
+        let request = RequestArgs {
+            title: None,
+            summary: None,
+            action: RequestArgsActions::Canister(RequestCanisterArgs {
+                action: RequestCanisterActionArgs::UpdateSettings(add_controller_args.clone()),
+            }),
+        }
+        .into_request(&dfx_orbit)
+        .await
+        .unwrap();
+        dfx_orbit.station.request(request.clone()).await.unwrap();
+
+        tokio::time::sleep(Duration::from_secs(1)).await;
+
+        let controllers = dfx_orbit.get_controllers(canister_id).await.unwrap();
+        assert_eq!(controllers, vec![user_test_id(1), canister_ids.station]);
```

Comment:
let's not check for list equality since the order of controllers could change, instead  we should check that the list has a length of two and contains both elements

Reply:
I added a small function that sorts both vecs before comparing them.